internal/cache/base.go

```diff
@@ -0,0 +1,207 @@
+package cache
+
+import (
+	"os"
+	"sync"
+	"time"
+
+	"fyne.io/fyne/v2"
+)
+
+var (
+	cacheDuration     = 1 * time.Minute
+	cleanTaskInterval = cacheDuration / 2
+
+	canvasRefreshCh = make(chan struct{}, 1)
+	expiredObjects  = make([]fyne.CanvasObject, 0, 50)
+	lastClean       time.Time
+)
+
+func init() {
+	if t, err := time.ParseDuration(os.Getenv("FYNE_CACHE")); err == nil {
+		cacheDuration = t
+		cleanTaskInterval = cacheDuration / 2
+	}
+}
+
+// CleanTask run cache clean task, it should be called on paint events.
+func CleanTask() {
+	now := time.Now()
+	// do not run clean task so fast
+	if now.Sub(lastClean) < 10*time.Second {
+		return
+	}
+	canvasRefresh := false
+	select {
+	case <-canvasRefreshCh:
+		canvasRefresh = true
+	default:
+		if now.Sub(lastClean) < cleanTaskInterval {
+			return
+		}
+	}
+	destroyExpiredSvgs(now)
+	if canvasRefresh {
+		// Destroy renderers on canvas refresh to avoid flickering screen.
+		destroyExpiredRenderers(now)
+		// canvases cache should be invalidated only on canvas refresh, otherwise there wouldn't
+		// be a way to recover them later
+		destroyExpiredCanvases(now)
+	}
+	lastClean = time.Now()
+}
+
+// ForceCleanFor forces a complete remove of all the objects that belong to the specified
+// canvas. Usually used to free all objects from a closing windows.
+func ForceCleanFor(canvas fyne.Canvas) {
```

Comment:
Is `Force` needed? perhaps `cache.CleanCanvas` would work?

Reply:
done in 6fc2bfc